IHP/View/CSSFramework.hs

```diff
@@ -199,6 +201,47 @@ instance Default CSSFramework where
                         </option>
                     |]
 
+
+            styledRadioFormField :: CSSFramework -> FormField -> Blaze.Html -> Blaze.Html
+            styledRadioFormField cssFramework@CSSFramework {styledInputClass, styledInputInvalidClass, styledFormFieldHelp} formField@FormField {fieldType, fieldName, placeholder, fieldLabel, fieldValue, fieldInputId, validatorResult, fieldClass, disabled, disableLabel, disableValidationResult, additionalAttributes, labelClass, required, autofocus } validationResult =
+                [hsx|
+                    {label}
+                    <fieldset>
+                        {forEach (options fieldType) (getRadio)}
+                    </fieldset>
+
+                    {validationResult}
+                    {helpText}
+                |]
+                where
+                    label = unless disableLabel [hsx|<label class={classes ["form-label", (labelClass, labelClass /= "")]} for={fieldInputId}>{fieldLabel}</label>|]
+                    inputInvalidClass = styledInputInvalidClass cssFramework formField
+                    helpText = styledFormFieldHelp cssFramework formField
+
+                    -- Get a single radio button.
+                    getRadio (optionLabel, optionValue) = [hsx|
+                        <div class="form-check">
+                            <input
+                                class={classes ["form-check-input", (inputInvalidClass, isJust validatorResult), (fieldClass, not (null fieldClass))]}
+                                type="radio"
+                                id={optionId}
+                                name={fieldName}
+                                value={optionValue}
+                                checked={optionValue == fieldValue}
+                                disabled={disabled}
+                                required={required}
+                                autofocus={autofocus}
+                                {...additionalAttributes}
+                            />
+                            {label}
+                        </div>
+                    |]
+                        where
+                            -- @todo: Need to find a way to guarantee that the id is unique and html valid.
+                            -- use `generateUniqueId`?
+                            optionId = fieldInputId <> "-" <> optionValue
```

Comment:
typically we use `_` instead of `-` for form Ids in IHP

Reply:
Fixed